Editor/EditorCore/EditorToolbar.cs

```diff
@@ -424,13 +424,28 @@ public void OnGUI()
                 }
             }
 
-            if ((e.type == EventType.Repaint || e.type == EventType.MouseMove) && !tooltipShown)
+            if ((evt.type == EventType.Repaint || evt.type == EventType.MouseMove) && !tooltipShown)
                 TooltipEditor.Hide();
 
-            if (e.type != EventType.Layout && !hovering)
+            if (evt.type != EventType.Layout && !hovering)
                 tooltipTimer.item1 = "";
 
-            if (forceRepaint || (EditorWindow.mouseOverWindow == this && e.delta.sqrMagnitude > .001f) || e.isMouse)
+            m_WantsRepaint |= EditorWindow.mouseOverWindow == window && evt.type == EventType.MouseMove;
+
+            if (Application.isFocused && m_WantsRepaint)
+                window.Repaint();
+
+            m_WantsRepaint = false;
+        }
+
+        void ScheduleRepaint()
+        {
+            m_WantsRepaint = true;
+        }
+
+        void ImmediateRepaint()
```

Comment:
Not sure about this name, at first I assumed it was RepaintImmediately and realised it was something else.